### PR TITLE
avm2: Use direct slot access in more native code

### PR DESCRIPTION
**Pull request by Lord-McSweeney**
This also fixes a bug with `BitmapData.paletteMap` not using the right coordinates for the destination point, and another bug with `PressAndTapGestureEvent::tapStageX`/`Y` not working at all.

Notable changes:
- `Object::is_instance_of` has been replaced with the new method `Value::is_instance_of`
- Uses of `FunctionObject::from_function` have been replaced with `FunctionObject::from_method`. Previously, the former allocated a prototype while the latter did not; now, `from_method` always allocates a prototype.

### CODE REVIEW

File: core/src/avm2/object/function_object.rs [hunk 1]
```diff
@@ -105,23 +105,6 @@ impl<'gc> FunctionObject<'gc> {
     ///
     /// This associated constructor will also create and initialize an empty
```
Comment (adrian17) [inline core/src/avm2/object/function_object.rs:106]: leftover docs

File: core/src/avm2/object.rs [hunk 2]
```diff
@@ -229,6 +234,16 @@ pub trait TObject<'gc>: 'gc + Collect + Debug + Into<Object<'gc>> + Clone + Copy
         self.base().get_property_local(name, activation)
     }
 
+    /// Same as get_property_local, but constructs a public Multiname for you.
+    fn get_string_property_local(
```
Comment (adrian17) [inline core/src/avm2/object.rs:238]: BTW, I think all three of these `*_string_property_local` could be `#[no_dynamic]`?

File: core/src/avm2/object.rs [hunk 3]
```diff
@@ -1429,6 +1402,7 @@ pub enum WeakObject<'gc> {
     FontObject(FontObjectWeak<'gc>),
     LocalConnectionObject(LocalConnectionObjectWeak<'gc>),
     SharedObjectObject(SharedObjectObjectWeak<'gc>),
+    SoundTransformObject(SoundTransformObjectWeak<'gc>),
```
Comment (adrian17) [inline core/src/avm2/object.rs:1405]: Any strong reason for doing this? I think I'd prefer this to keep it on the AS3 side, especially if others like `Point` are also on the AS3 side.

File: core/src/avm2/globals/flash/display/Shader.as [hunk 4]
```diff
@@ -4,7 +4,9 @@ package flash.display {
     import flash.utils.ByteArray;
 
     public class Shader {
+        [Ruffle(InternalSlot)]
```
Comment (adrian17) [inline core/src/avm2/globals/flash/display/Shader.as:7]: (note: some shader-related changes got mixed in with the Rectangle commit)

File: core/src/avm2/object.rs (hunk 3 above)
Comment (Lord-McSweeney) [reply core/src/avm2/object.rs:1405]: There aren't any accessible slots, unlike in Point, and we need to access the properties from `display_object.rs`, so I thought this worked better than the repeated `get_slot`s in there.

**Review by adrian17 — APPROVED** [review]

**Pull request merged**